lib/ts/BufferWriter.h

```diff
@@ -681,6 +708,24 @@ operator<<(BufferWriter &w, int const &i)
   return bwformat(w, BWFSpec::DEFAULT, static_cast<intmax_t>(i));
 }
 
+// std::string support
+/** Print to a @c std::string
+
+    Print to the string @a s. If there is overflow then resize the string sufficiently to hold the output
+    and print again. The effect is the string is resized only as needed to hold the output.
+ */
+template <typename... Rest>
+void
+bwprint(std::string &s, ts::TextView fmt, Rest &&... rest)
+{
+  auto len = s.size();
+  size_t n = ts::FixedBufferWriter(const_cast<char *>(s.data()), s.size()).print(fmt, std::forward<Rest>(rest)...).extent();
+  s.resize(n);   // always need to resize - if shorter, must clip pre-existing text.
```

Comment:
The comment "always need to resize" is a bit confusing since the comment above says resized only as needed.